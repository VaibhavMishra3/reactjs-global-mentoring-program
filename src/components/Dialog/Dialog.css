.dialog-overlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.8);
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 1000;
}

.dialog-content {
  background-color: #232323;
  padding: 20px;
  width: 500px;
  border-radius: 8px;
  color: #fff;
  position: relative;
}

.dialog-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 20px;
}

.dialog-close {
  background: none;
  border: none;
  font-size: 24px;
  color: #fff;
  cursor: pointer;
}

.dialog-body {
  margin-top: 10px;
}
