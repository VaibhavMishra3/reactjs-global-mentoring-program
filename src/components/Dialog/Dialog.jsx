import React from 'react';
import { PropTypes } from 'prop-types';
import { Portal } from 'react-portal';
import { Modal, CloseButton } from 'react-bootstrap';

const Dialog = ({ title, children, handleClose }) => {
  return (
    <Portal>
      <Modal show={true} centered size='lg'>
        <Modal.Header>
          <Modal.Title>
            {title}
          </Modal.Title>
          <CloseButton onClick={handleClose} />
        </Modal.Header>
        <div>
          {children}
        </div>
      </Modal>
    </Portal>
  );
}

Dialog.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node).isRequired,
    PropTypes.node.isRequired
  ]).isRequired,
  handleClose: PropTypes.func.isRequired
};

export default Dialog;
