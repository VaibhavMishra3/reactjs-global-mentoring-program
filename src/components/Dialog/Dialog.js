import React from "react";
import ReactDOM from "react-dom";
import FocusTrap from "focus-trap-react";
import "./Dialog.css"; // You can add styles here or inline

function Dialog({ title, children, onClose }) {
  return ReactDOM.createPortal(
    <FocusTrap>
      <div className="dialog-overlay">
        <div className="dialog-content">
          <header className="dialog-header">
            <h2>{title}</h2>
            <button className="dialog-close" onClick={onClose}>
              Ã—
            </button>
          </header>
          <div className="dialog-body">{children}</div>
        </div>
      </div>
    </FocusTrap>,
    document.body
  );
}

export default Dialog;
