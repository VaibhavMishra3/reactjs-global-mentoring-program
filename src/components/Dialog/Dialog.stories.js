import Dialog from './Dialog';

export default {
  component: Dialog
};

export const Default = {
  args: {
    title: 'Title',
    children: 'children',
    handleClose: (event) => console.log('dialog closed')
  },
  parameters: {
  }
};
