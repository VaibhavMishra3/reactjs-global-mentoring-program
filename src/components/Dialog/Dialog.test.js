import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { cleanup, render, screen, fireEvent } from '@testing-library/react'

import Dialog from './Dialog';

describe('Dialog', () => {

  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element, _node) => {
      return element;
    });
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
    cleanup();
  });

  it('should match snapshot', () => {
    const dialog = renderer
      .create(
        <Dialog
          title='Title'
          children='children'
          handleClose={console.log}
        />
      )
      .toJSON();

    expect(dialog).toMatchSnapshot();
  });

  it('render correctly', () => {
    render(
      <Dialog
        title='Title'
        children='children'
        handleClose={console.log}
      />
    );

    const closeButton = screen.getByRole('button');
    expect(closeButton).toBeInTheDocument();
    expect(closeButton.tagName).toBe('BUTTON');
    expect(closeButton.className).toBe('btn-close');

    const title = screen.getByText('Title');
    expect(title).toBeInTheDocument();
    expect(title.tagName).toBe('DIV');
    expect(title.className).toBe('modal-title h4');

    const children = screen.getByText('children');
    expect(children).toBeInTheDocument();
    expect(children.tagName).toBe('DIV');
    expect(children.className).toBe('');
  });

  it('should call the close handler', () => {
    const handleCloseMock = jest.fn();

    render(
      <Dialog
        title='Title'
        children='children'
        handleClose={handleCloseMock}
      />
    );
    const closeButton = screen.getByRole('button');

    fireEvent.click(closeButton, {});

    expect(handleCloseMock).toHaveBeenCalled();
  });

});
