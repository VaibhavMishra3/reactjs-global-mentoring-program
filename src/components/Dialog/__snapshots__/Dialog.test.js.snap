// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should match snapshot 1`] = `
Array [
  <div
    className="fade modal-backdrop show"
    onClick={[Function]}
  />,
  <div
    aria-modal={true}
    className="fade modal show"
    onClick={[Function]}
    onMouseUp={[Function]}
    role="dialog"
    style={
      Object {
        "display": "block",
      }
    }
    tabIndex={-1}
  >
    <div
      className="modal-dialog modal-lg modal-dialog-centered"
      onMouseDown={[Function]}
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <div
            className="modal-title h4"
          >
            Title
          </div>
          <button
            aria-label="Close"
            className="btn-close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div>
          children
        </div>
      </div>
    </div>
  </div>,
]
`;
